src/lib/viewers/archive/ArchiveViewer.js

```diff
@@ -0,0 +1,348 @@
+                {
+                    type: 'folder',
+                    absolute_path: 'test/level-1/level-2/',
+                    name: 'level-2',
+                },
+            ],
+        },
+        parent: 'level-2',
+        item_collection: null,
+    },
+];
+
+const virtualizedTestData = [
+    {
+        id: 0,
+        name: "01 - Adam Sanders - Life I'm Livin'.m4a",
+        modified_at: '2019-05-10T18:02:30Z',
+        size: 6823316,
+        type: 'file',
+        modified_by: {
+            id: '666',
+            name: 'Mingze',
+            email: 'mxiao@box.com',
+            login: 'mxiao',
+        },
+        path: {
+            id: 0,
+            name: "01 - Adam Sanders - Life I'm Livin'.m4a",
+            size: 6823316,
+            itemPath: [{ name: 'folder1' }, { name: 'folder2' }],
+            itemType: 'file',
+        },
+    },
+    {
+        id: 1,
+        name: 'coraline.mp4',
+        modified_at: '2013-02-19T19:56:50Z',
+        size: 4464268,
+        type: 'file',
+        path: {
+            id: 1,
+            name: 'coraline.mp4',
+            size: 4464268,
+            itemPath: [{ id: '0' }],
+            itemType: 'file',
+        },
+    },
+];
+
+class ArchiveViewer extends BaseViewer {
+    /**
+     * @inheritdoc
+     */
+    setup() {
+        if (this.isSetup) {
+            return;
+        }
+
+        // Call super() first to set up common layout
+        super.setup();
+
+        this.archiveEl = this.createViewer(document.createElement('div'));
+        this.archiveEl.className = 'bp-archive';
+    }
+
+    /**
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        if (this.archiveComponent) {
+            this.archiveComponent.destroy();
+        }
+        super.destroy();
+    }
+
+    /**
+     * Loads a archive file.
+     *
+     * @return {void}
+     */
+    load() {
+        super.load();
+
+        return Promise.all([this.loadAssets(JS, CSS), this.getRepStatus().getPromise()])
+            .then(this.postLoad)
+            .catch(this.handleAssetError);
+    }
+
+    /**
+     * Loads archive file representation.
+     *
+     * @private
+     * @return {Promise} promise to get archive content
+     */
+    postLoad = () => {
+        const { representation } = this.options;
+        const template = representation.content.url_template;
+        const contentUrl = this.createContentUrlWithAuthParams(template);
+        this.startLoadTimer();
+
+        this.api
+            .get(contentUrl)
+            .then(data => {
+                if (this.isDestroyed()) {
+                    return;
+                }
+
+                this.data = testData;
+
+                this.finishLoading();
+            })
+            .catch(this.handleAssetError);
+    };
+
+    /**
+     * Finishes loading the archive data
+     *
+     * @private
+     * @return {void}
+     */
+    finishLoading() {
+        /* global BoxArchive */
```

Comment:
It's not really clear why this needs to be a global. Can you add a comment with an explanation?